resource/pod.go

```diff
@@ -36,14 +36,20 @@ type PodResource struct {
 }
 
 // GetPodResource extracts a PodResource from an AdmissionRequest
-func GetPodResource(ar *admissionv1beta1.AdmissionRequest, ctx context.Context) *PodResource {
+func GetPodResource(ctx context.Context, ar *admissionv1beta1.AdmissionRequest) *PodResource {
 	c := GetResourceCache(ctx)
 	return c.getOrSet(cacheKeyPod, func() interface{} {
 		return decodePodResource(ar)
 	}).(*PodResource)
 }
 
 func decodePodResource(ar *admissionv1beta1.AdmissionRequest) *PodResource {
+	// omit Pod exec
+	switch ar.Kind {
+	case metav1.GroupVersionKind{Group: "", Version: "v1", Kind: "PodExecOptions"}:
```

Comment:
Should PodAttachOptions be included as well? https://github.com/golang/build/blob/master/kubernetes/api/types.go#L1964

Reply:
The webhook currently does not accept that, but that could change in the future. Hmm, possibly `PodLogOptions` and `PodProxyOptions` too. They might all have a metav1.GroupVersionResource that matches the switch below. It'd be worth opening up the webhook and messing with k-rail in debug mode to ensure we catch these.

The other action item on my todo is to add smoke tests on a `kind` cluster in CI which can catch these cases if we test for them.